src/gui/editors/PianoRoll.cpp

```diff
@@ -961,6 +961,8 @@ void PianoRoll::shiftSemiTone( int amount ) // shift notes by amount semitones
 		}
 	}
 
+	m_pattern->dataChanged();
```

Comment:
I think you need to call ```m_pattern->rearrangeAllNotes();``` before `dataChanged()` as you may have moved the note past another note so they need to be sorted again.